drivers/include/pn532.h

```diff
@@ -0,0 +1,327 @@
+ */
+typedef struct {
+    const pn532_params_t *conf;
+    mutex_t trap;
+} pn532_t;
+
+/**
+ * @brief Internal buffer size
+ *
+ * A small buffer size is enough for most applications, however if large NDEF
+ * files are to be written this size shall be increased. Otherwise the files
+ * can be written in chunks.
+ */
+#ifndef PN532_BUFFER_LEN
+#define PN532_BUFFER_LEN     64
+#endif
+
+/**
+ * @brief Helpers to extract firmware information from word
+ * @{
+ */
+#define PN532_IC_VERSION(fwver)  ((fwver >> 24) & 0xff)
+#define PN532_FW_VERSION(fwver)  ((fwver >> 16) & 0xff)
+#define PN532_FW_REVISION(fwver) ((fwver >>  8) & 0xff)
+#define PN532_FW_FEATURES(fwver) ((fwver) & 0xff)
+/** @} */
+
+/**
+ * @brief Possible SAM configurations
+ */
+typedef enum {
+    PN532_SAM_NORMAL = 1,
+    PN532_SAM_VIRTUAL,
+    PN532_SAM_WIRED,
+    PN532_SAM_DUAL
+} pn532_sam_conf_mode_t;
+
+/**
+ * @brief PN532 supported targets
+ */
+typedef enum {
+    PN532_BR_106_ISO_14443_A = 0,
+    PN532_BR_212_FELICA,
+    PN532_BR_424_FELICA,
+    PN532_BR_106_ISO_14443_B,
+    PN532_BR_106_JEWEL
+} pn532_target_t;
+
+/**
+ * @brief ISO14443A Card types
+ */
+typedef enum {
+    ISO14443A_UNKNOWN,
+    ISO14443A_MIFARE,
+    ISO14443A_TYPE4
+} nfc_iso14443a_type_t;
+
+/**
+ * @brief ISO14443A tag description
+ */
+typedef struct {
+    char sns_res;
+    char sel_res;
+    char target;
+    char auth;
+    char id_len;
+    nfc_iso14443a_type_t type;
+    char id[8];
+} nfc_iso14443a_t;
+
+/**
+ * @brief Mifare keys
+ */
+typedef enum {
+    PN532_MIFARE_KEY_A = 0x60,
+    PN532_MIFARE_KEY_B = 0x61
+} pn532_mifare_key_t;
+
+/**
+ * @brief Obtain Tag 4 data length from buffer
+ *
+ * This is useful in case the length has been read and one intents to read the
+ * data.
+ */
+#define PN532_ISO14443A_4_LEN_FROM_BUFFER(b) ((b[0] << 8) | b[1])
+
+/**
+ * @brief Hard reset the chipset
+ *
+ * The chipset is reset by toggling the reset pins
+ *
+ * @param[in]  dev          target device
+ *
+ */
+void pn532_reset(pn532_t *dev);
+
+/**
+ * @brief Initialize the module and peripherals
+ *
+ * This is the first method to be called in order to interact with the pn532.
+ * It configures the GPIOs and the i2c interface.
+ *
+ *  @param[in]  dev         target device
+ *  @param[in]  params      configuration parameters
+ *
+ * @return                  0 on success
+ */
+int pn532_init(pn532_t *dev, const pn532_params_t *params);
+
+/**
+ * @brief Get the firmware version of the pn532
+ *
+ * The firmware version returned is a 4 byte long value:
+ *  - ic version,
+ *  - fw version,
+ *  - fw revision
+ *  - feature support
+ *
+ * @param[in]  dev          target device
+ * @param[out] fw_ver       encoded firmware version
+ *
+ * @return                  0 on success
+ */
+int pn532_fw_version(pn532_t *dev, uint32_t *fw_ver);
+
+/**
+ * @brief Read register of the pn532
+ *
+ * Refer to the datasheet for a comprehensive list of registers and meanings.
```

Comment:
it's always dangerous to add links to external servers since they might change... but it makes sense. If it works it works, if it doesn't then one has to search for it :-) :+1: